src/appleseed/renderer/modeling/aov/cryptomatteaov.cpp

```diff
@@ -510,11 +510,17 @@ namespace
                         pixel_values.push_back(b);
 
                         // Remove background contribution.
+                        size_t ranked_vector_start = 0;
+                        size_t ranked_vector_size = ranked_vector.size();
+
                         if (ranked_vector.size() > 1 && m3hash_preview == 0)
-                            ranked_vector.erase(ranked_vector.begin());
+                        {
+                            ranked_vector_start = 1;
+                            ranked_vector_size -= 1;
+                        }
 
                         // Ranked channels.
-                        for (size_t i = 0, e = min(ranked_vector.size(), m_num_layers); i < e; ++i)
+                        for (size_t i = ranked_vector_start, e = min(ranked_vector.size(), m_num_layers); i < e; ++i)
```

Comment:
This is still using `ranked_vector.size()` here, is it intended?

Reply:
Yes

Reply:
More thorough: we're skipping over the first ranked vector but we still need to iterate over rest of them.  If I used 'ranked_vecotr_size' in this context it would stop at the last one and not iterate over it.